### PR TITLE
Resolving GitHub Issues & Overall Improvements 

### PR DESCRIPTION
**Pull request by tiutiunnykleasoft**
* Resolved issue, when payment method details does force connection to any transaction.

### CODE REVIEW

File: src/Entities/Order.php [hunk 1]
```diff
@@ -120,6 +120,8 @@ public function getPaymentUrl(): string
         return $this->getCurrentTransaction()->getPaymentUrl()->get();
     }
 
+    //@TODO: Add get order url for pay now payment method.
+
     public function getWebhookUrl(): BaseField
```
Comment (BohdanLeasoft) [inline src/Entities/Order.php:125]: Merchant order id isn't displaying on portal. It look like he filling in order in different format. After creating Order with `merchantOrderId`, I got something like this:
```
 ["merchantOrderId"]=>
  string(7) "122"
```
  

In this file on line 52-55 placed this part of code:
```
        if ($merchantOrderId) $this->merchantOrderId = $this->createSimpleField(
            propertyName: 'merchantOrderId',
            value: $merchantOrderId
        );
```
Problem was solved and Merchant order id started displaying on portal, when i replaced `merchantOrderId` to `merchant_order_id`

**Comment by BohdanLeasoft** [comment]
Is it possible to extend method `getValidOrder()` in class [Tests/OrderStub.php](https://github.com/gingerpayments/ginger-php-plugin-sdk/blob/9088dea53e2b50241fdbec2255b6774726912d40/src/Tests/OrderStub.php) to cover such properties like: merchantOrderId, returnUrl and webhookUrl?

Usually, in plugins orders has such properties. So, with them test order will be closer to plugin order

For now `getValidOrder` looks like this:
```
    static function getValidOrder(): Order
    {
        return new Order(
            currency: new Currency('EUR'),
            amount: new Amount(500),
            transactions: self::getValidTransactions(),
            customer: self::getValidCustomer(),
            orderLines: self::getValidOrderLines(),
            extra: self::getValidExtra(),
            client: self::getValidClient(),
            description: 'GingerPluginSDKAutomaticTest'
        );
    }
```

**Comment by BohdanLeasoft** [comment]
I just created new [issue](https://github.com/gingerpayments/ginger-php-plugin-sdk/issues/19) about min value of `vat_percentage `. Could you have a look?

It would be great to change that too

**Comment by tiutiunnykleasoft** [comment]
> Is it possible to extend method `getValidOrder()` in class [Tests/OrderStub.php](https://github.com/gingerpayments/ginger-php-plugin-sdk/blob/9088dea53e2b50241fdbec2255b6774726912d40/src/Tests/OrderStub.php) to cover such properties like: merchantOrderId, returnUrl and webhookUrl?
> 
> Usually, in plugins orders has such properties. So, with them test order will be closer to plugin order
> 
> For now `getValidOrder` looks like this:
> 
> ```
>     static function getValidOrder(): Order
>     {
>         return new Order(
>             currency: new Currency('EUR'),
>             amount: new Amount(500),
>             transactions: self::getValidTransactions(),
>             customer: self::getValidCustomer(),
>             orderLines: self::getValidOrderLines(),
>             extra: self::getValidExtra(),
>             client: self::getValidClient(),
>             description: 'GingerPluginSDKAutomaticTest'
>         );
>     }
> ```

You could convert order object to array and merge another array to it.

**Pull request merged**